crates/primitives/src/alloy_compat.rs

```diff
@@ -227,7 +233,31 @@ impl TryFrom<alloy_rpc_types::Transaction> for Transaction {
                 }))
             }
             #[cfg(feature = "optimism")]
-            Some(TxType::Deposit) => todo!(),
+            Some(TxType::Deposit) => {
+                Ok(Self::Deposit(TxDeposit {
+                    // TODO: Uses the wrong error here, need alloy ConversionError enum updated with new errors for this.
```

Comment:
do you want to add a `Custom(String)` variant to the alloy error?

Reply:
i think best to just add `ConversionError::MissingSourceHash`, `ConversionError::InvalidSourceHash`, and `ConversionError::InvalidMintValue` for consistency.